ng/manage.ts

```diff
@@ -436,6 +436,72 @@ angular.module("armExplorer", ["ngRoute", "ngAnimate", "ngSanitize", "ui.bootstr
         window.location.href = "api/tenants/" + $scope.selectedTenant.id;
     };
 
+    $scope.resourceSearch = () => {
+        // trigger a refresh if needed, but do not need to wait for it to finish since cache should already exist
+        refreshResourceSearchCache();
+
+        var results: IResourceSearchSuggestion[] = [];
+        var keyword = $scope.resourceSearchModel.searchKeyword || "";
+        var stringCompare = (strA: string, strB: string): number=> {
```

Comment:
yes, this one is debatable. 

.localCompare will use the browser locale. similar to "toLocalUpperCase".   reason i do not use it is because resource name and type are all english (if i am not wrong, since they are part of csm id)

if we use .localeCompare, different region of user might see the list in different order (might not be a big deal i think). so i decide to do our own string compare instead.